contracts/cw1-whitelist/src/contract.rs

```diff
@@ -0,0 +1,198 @@
+use anyhow::Error;
+use cosmwasm_std::{from_slice, Addr, Binary, Deps, DepsMut, Empty, Env, MessageInfo, Response};
+use cw1::msg::{ExecMsg, QueryMsg};
+use cw1::Cw1;
+use cw1::*;
+use cw_storage_plus::Map;
+use sylvia::contract;
+
+use self::contract::InstantiateMsg;
+
+pub struct Cw1WhitelistContract {
+    members: Map<'static, Addr, Empty>,
+}
+
+impl Cw1 for Cw1WhitelistContract {
+    type Error = Error;
+    fn add_member(
+        &self,
+        (deps, _env, _info): (
+            cosmwasm_std::DepsMut,
+            cosmwasm_std::Env,
+            cosmwasm_std::MessageInfo,
+        ),
+        member: String,
+    ) -> Result<Response, Self::Error> {
+        self.members
+            .save(deps.storage, deps.api.addr_validate(&member)?, &Empty {})?;
+
+        Ok(Response::new())
+    }
+
+    fn find_member(
+        &self,
+        (deps, _env): (cosmwasm_std::Deps, cosmwasm_std::Env),
+        member: String,
+    ) -> Result<FindMemberResponse, Self::Error> {
+        let is_present = self
+            .members
+            .has(deps.storage, deps.api.addr_validate(&member)?);
+
+        Ok(FindMemberResponse { is_present })
+    }
+}
+
+#[contract(module=contract, error=Error)]
+#[messages(msg as Cw1)]
+impl Cw1WhitelistContract {
+    pub const fn new() -> Self {
+        Self {
+            members: Map::new("members"),
+        }
+    }
+
+    #[msg(instantiate)]
+    pub fn instantiate(
+        &self,
+        (deps, _env, _msg): (DepsMut, Env, MessageInfo),
+        members: Vec<String>,
+    ) -> Result<Response, Error> {
+        for addr in members.into_iter() {
+            self.members
+                .save(deps.storage, deps.api.addr_validate(&addr)?, &Empty {})?;
+        }
+
+        Ok(Response::new())
+    }
+
+    pub(crate) fn entry_instantiate(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: &[u8],
+    ) -> Result<Response, Error> {
+        let msg: InstantiateMsg = from_slice(msg)?;
+        msg.dispatch(self, (deps, env, info))
+    }
+
+    pub(crate) fn entry_execute(
+        &self,
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: &[u8],
+    ) -> Result<Response, Error> {
+        let msg: ExecMsg = from_slice(msg)?;
+        msg.dispatch(self, (deps, env, info))
+    }
+
+    pub(crate) fn entry_query(&self, deps: Deps, env: Env, msg: &[u8]) -> Result<Binary, Error> {
+        let msg: QueryMsg = from_slice(msg)?;
+        msg.dispatch(self, (deps, env))
+    }
```

Comment:
I know I did this in the original `cw1-whitelist`, but I feel this is a very nasty boilerplate. Those functions adds more complexity than just calling `from_slice::<ExecMsg>(msg)?.dispatch(CONTRACT, (deps, env, info))` in the entry point - I think we should prefer this approach. I know you would double it in `Contract` implementation, but this one is to be generated anyway so you will not see that.